chart/templates/deployment-sidekiq.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "mastodon.fullname" . }}-sidekiq
+  labels:
+    {{- include "mastodon.labels" . | nindent 4 }}
+spec:
+{{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+{{- end }}
+  selector:
+    matchLabels:
+      {{- include "mastodon.selectorLabels" . | nindent 6 }}
+      component: rails
+  template:
+    metadata:
+    {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+        # roll the pods to pick up any db migrations
+        rollme: {{ randAlphaNum 5 | quote }}
+    {{- end }}
+      labels:
+        {{- include "mastodon.selectorLabels" . | nindent 8 }}
+        component: rails
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "mastodon.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      # ensure we run on the same node as the other rails components; only
+      # required when using PVCs that are ReadWriteOnce
```

Comment:
Does this mean that all sidekiq instances have to live together on one node? Doesn’t that severely limit scaling?

Reply:
Is there any way around that, without `ReadWriteMany` PVCs?  (The cluster I tested this on is on Linode, which doesn't yet offer rwx storage.)

Reply:
One way to mitigate it is something like https://github.com/tootsuite/mastodon/pull/14090/commits/514c14539fa32e751890b2c0cbf16a7a91db41d4, although that means you can only specify one access mode for the PVCs; I'm not sure how much of a drawback that is.

Reply:
This sort of shared write space really requires ReadWriteMany to get any kind of scaling and since a lot of providers have this out of the box or allow you to configure NFS easily, I would think supporting it is worth while and that forcing ReadWriteOnce and the affinity is going to cause confusion.

Reply:
In addition to https://github.com/tootsuite/mastodon/commit/514c14539fa32e751890b2c0cbf16a7a91db41d4, does https://github.com/tootsuite/mastodon/pull/14090/commits/4612f39a73adc4eac9944382dde0963c49fbc4d8 clarify things enough?

Reply:
Rebased those commits as b97babf.